Add --strict flag for returning exit code 1 on warnings

LeoVerto opened the pull request:
Fixes #8.

figuerascarlos commented on cmd/root.go:
```diff
@@ -128,9 +130,9 @@ func outputResult(results []validate.ValidationResult) {
-func containsError(results []validate.ValidationResult) bool {
+func containsSeverity(results []validate.ValidationResult, severity validate.Severity) bool {
 	for _, result := range results {
-		if result.Severity == validate.SeverityError {
```
```go
switch result.Severity {     
case validate.SeverityError:  
    return true
case validate.SeverityWarning:
   return strict
default :
return false
}
```

figuerascarlos commented on cmd/root.go:
```diff
@@ -106,7 +108,7 @@ func runValidate(filenames []string, schema string, crds []string, recursive boo
 			validationResults := fileValidator.Validate(fileBytes)
 			outputResult(validationResults)
-			if containsError(validationResults) {
+			if containsSeverity(validationResults, validate.SeverityError) || (strict && containsSeverity(validationResults, validate.SeverityWarning)) {
```
```go
if containsSeverity(validationResults) {
```

figuerascarlos reviewed (changes requested):
Really good, just a small change in the validation to avoid execute the function twice and make a little bit more readable the check of the severity. 

A lot thanks for the PR and collaborate with this project.

LeoVerto commented:
Thank you for the feedback @figuerascarlos! I've added your suggestion and added some additional tests.

I noticed I messed up your indentation style somewhere along the way but that should be fixed now.

There is however one thing that has me stumped, maybe it's because I'm a novice to Go but wouldn't this piece of code
falsely return `false` on a Warning if strict mode is deactivated? And if it does, why doesn't my `test unknown crd then invalid yaml` test case catch it doing that?
```go
for _, result := range results {
	switch result.Severity {
	case validate.SeverityError:
		return true
	case validate.SeverityWarning:
		return strict
	}
}
```

figuerascarlos commented:
You are absolutely right, can you add in the case of warning and of checking the flag.
```go
case validate.SeverityWarning:
if strict {
 return true
}
return false
```

Sorry for the big error in the suggestion

fllaca commented:
@LeoVerto your test is not catching the case you spotted because it would happen when the error is in the same YAML file.  `containsSeverity()` is executed for every analyzed file, so it gets executed twice in your `test unknown crd then invalid yaml`, capturing the validation error when checking the second file. 

I think testing with this config  (we need to add it to `cmd/testdata/manifests`) containing a WARN and an ERROR in the same YAML file would reproduce and catch that case:

```yaml
---
apiVersion: example.io/v1
kind: UnknownCRD
metadata:
  name: example-unknown-kind
  namespace: example
spec:
  someKey: someValue
---
this
 is:
    wrong
  yaml
```
(expected results are 1 WARN and 1 ERROR, and exit code 1 in strict mode and 0 when non-strict mode).

good catch btw!

fllaca commented:
hi @LeoVerto !

> I'd be happy to contribute again in the future!

That would be awesome! If feeling fancy you can go to the issues tab of the repo and you can pick any of them, looking forward to your PRs! :D 

We just published `v0.0.1-rc2` bundling your feature from this PR and some other fixes, here you have it: https://github.com/fllaca/scheriff/releases/tag/v0.0.1-rc2, I hope it serves you well!
